src/Shared/ServerInfrastructure/ManualResetValueTaskSource.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading.Tasks.Sources;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal
+{
+    internal sealed class ManualResetValueTaskSource<T> : IValueTaskSource<T>, IValueTaskSource
```

Comment:
Why can't we use the version of this in the runtime? https://github.com/dotnet/runtime/blob/master/src/libraries/Common/tests/System/Threading/Tasks/Sources/ManualResetValueTaskSource.cs

Reply:
Its in a runtime test project